include/actor_types.h

```diff
@@ -5,7 +5,121 @@
+#define ACTOR_WARIO_STADIUM_SIGN  0x23
+#define ACTOR_PADDLE_WHEEL_BOAT   0x26
+#define ACTOR_RAILROAD_CROSSING   0x27
+#define ACTOR_BLUE_SPINY_SHELL    0x2A
+#define ACTOR_HAB_ITEM_BOX        0x2B // HAB -> Hot Air Baloon
+#define ACTOR_KIWANO_FRUIT        0x2D
+
+/*
+Used by the locomotive, tender, and passenger car
+*/
+struct TrainCar {
+    /* 0x00 */ s16 type;
+    /* 0x02 */ s16 flags;
+    /* 0x04 */ s16 unk_04;
+    /* 0x06 */ s16 wheelRot;
+    /* 0x08 */ f32 unk_08;
+    /* 0x0C */ f32 unk_0C;
+    /* 0x10 */ Vec3s rot;
+    /* 0x10 */ s16 unk_16;
+    /* 0x18 */ Vec3f pos;
+    /* 0x24 */ Vec3f velocity;
+    /* 0x30 */ UnkActorInner unk30;
+}; // size = 0x70
+
+struct RailroadCrossing {
+    /* 0x00 */ s16 type;
+    /* 0x02 */ s16 flags;
+    /* 0x04 */ s16 someTimer;
+    /* 0x06 */ s16 crossingId;
+    /* 0x08 */ f32 unk_08;
+    /* 0x0C */ f32 unk_0C;
+    /* 0x10 */ Vec4s unk_10;
+    /* 0x18 */ Vec3f pos;
+    /* 0x24 */ Vec3f velocity;
+    /* 0x30 */ UnkActorInner unk30;
+}; // size = 0x70
+
+struct FallingRock {
+    /* 0x00 */ s16 type;
+    /* 0x02 */ s16 flags;
+    /* 0x04 */ s16 respawnTimer;
+    /* 0x06 */ s16 unk_06;
+    /* 0x08 */ f32 unk_08;
+    /* 0x0C */ f32 boundingBoxSize;
+    /* 0x10 */ Vec3s rot;
+    /* 0x16 */ s16 unk_16;
+    /* 0x18 */ Vec3f pos;
+    /* 0x24 */ Vec3f velocity;
+    /* 0x30 */ UnkActorInner unk30;
+}; // size = 0x70
+
+struct YoshiValleyEgg {
+    /* 0x00 */ s16 type;
+    /* 0x02 */ s16 flags;
+    /* 0x04 */ s16 unk_04;
+    /* 0x06 */ s16 unk_06;
+    /* 0x08 */ f32 pathRadius;
+    /* 0x0C */ f32 boundingBoxSize;
+    /* 0x10 */ s16 pathRot;
+    /* 0x12 */ s16 eggRot;
+    /* 0x14 */ s16 unk_14;
+    /* 0x16 */ s16 unk_16;
+    /* 0x18 */ Vec3f pos;
+    // Note, pathCenter[1] should be understood to be the Y velocity of the egg
+    // pathCenter[0] and pathCenter[2] are the X,Z coordinates of the center of the path
+    /* 0x24 */ Vec3f pathCenter;
+    /* 0x30 */ UnkActorInner unk30;
+}; // size = 0x70
+
+struct KiwanoFruit {
+    /* 0x00 */ s16 type;
+    /* 0x02 */ s16 flags;
+    /* 0x04 */ s16 targetPlayer; // Id of the player this actor tracks. Each player has their own kiwano actor just for them
+    /* 0x06 */ s16 state;
+    /* 0x08 */ f32 bonkTimer; // bonkState? Not sure what this is tracking, but its some form of count down that starts after the fruit hits you
+    /* 0x0C */ f32 boundingBoxSize;
+    /* 0x10 */ s16 animState;
+    /* 0x12 */ s16 animTimer;
+    /* 0x14 */ s16 unk_14;
+    /* 0x16 */ s16 unk_16;
+    /* 0x18 */ Vec3f pos;
+    /* 0x24 */ Vec3f velocity;
+    /* 0x30 */ UnkActorInner unk30;
+}; // size = 0x70
+
+struct PaddleWheelBoat {
+    /* 0x00 */ s16 type;
+    /* 0x02 */ s16 flags;
+    /* 0x04 */ s16 unk_04;
+    /* 0x06 */ s16 wheelRot;
+    /* 0x08 */ f32 unk_08;
+    /* 0x0C */ f32 unk_0C;
+    /* 0x10 */ Vec3s boatRot; // Weirdly, its boatRot[1] (Y rotation?) that changes the direction the boat faces
```

Comment:
xyz. Isn't that pretty normal?

Reply:
I should have removed that comment. I think I know what's going on and its a matter of the rotation stuff being a little different than I expected. My presumption was that the `rot[Y]` value changing would mean the object rotates in the Y direction, but it appears that it actually means the that object rotates around the Y axis. So its entirely sensible I was just misunderstanding what the `rot` actually represented

Reply:
Ah. rotAround? Might be q better term. rotTranslate?